crates/egui/src/context.rs

```diff
@@ -1610,30 +1634,93 @@ impl Context {
     ///     style.spacing.item_spacing = egui::vec2(10.0, 20.0);
     /// });
     /// ```
-    pub fn style_mut(&self, mutate_style: impl FnOnce(&mut Style)) {
-        self.options_mut(|opt| mutate_style(std::sync::Arc::make_mut(&mut opt.style)));
+    pub fn style_mut(&self, mut mutate_style: impl FnMut(&mut Style)) {
+        self.options_mut(|opt| {
+            mutate_style(std::sync::Arc::make_mut(&mut opt.dark_style));
+            mutate_style(std::sync::Arc::make_mut(&mut opt.light_style));
+        });
+    }
+
+    /// The [`Style`] used by all subsequent windows, panels etc in dark mode.
+    pub fn dark_style(&self) -> Arc<Style> {
+        self.options(|opt| opt.dark_style.clone())
+    }
+
+    /// Mutate the [`Style`] used by all subsequent windows, panels etc in dark mode.
+    ///
+    /// Example:
+    /// ```
+    /// # let mut ctx = egui::Context::default();
+    /// ctx.dark_style_mut(|style| {
+    ///     style.spacing.item_spacing = egui::vec2(10.0, 20.0);
+    /// });
+    /// ```
+    pub fn dark_style_mut(&self, mutate_style: impl FnOnce(&mut Style)) {
+        self.options_mut(|opt| mutate_style(std::sync::Arc::make_mut(&mut opt.dark_style)));
     }
 
-    /// The [`Style`] used by all new windows, panels etc.
+    /// The [`Style`] used by all new windows, panels etc. in dark mode.
+    /// Use [`Self::set_theme`] to choose between dark and light mode.
     ///
-    /// You can also change this using [`Self::style_mut`]
+    /// You can also change this using [`Self::dark_style_mut`]
     ///
     /// You can use [`Ui::style_mut`] to change the style of a single [`Ui`].
-    pub fn set_style(&self, style: impl Into<Arc<Style>>) {
-        self.options_mut(|opt| opt.style = style.into());
+    pub fn set_dark_style(&self, style: impl Into<std::sync::Arc<crate::Style>>) {
```

Comment:
To avoid having too large of an API surface, I suggest we combine `set_dark_style` and `set_light_style` with `ctx.set_style(theme, style)`. Same with the getters.

Reply:
Good idea! I had to rename the current `style` method to `active_style` as a consequence (this broke a lot of places, but I think `active_style` is a lot clearer than just `style`.
I also had to rename the currently named `style_mut` (which mutates both dark and light style) to `styles_mut`.